.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,127 @@
+      branch:
+        description: 'Branch name'
+        required: true
+        default: 'develop'
+
+jobs:
+  # Each event type has a different variable for the branch name
+  get-branch:
+    name: Get branch name
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.branch.outputs.branch }}
+    steps:
+      - name: Branch used
+        id: branch
+        run: |
+          if [[ "${GITHUB_EVENT_NAME}" == "push" ]]; then
+            echo "::set-output name=branch::${{ github.ref_name  }}"
+          elif [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
+            echo "::set-output name=branch::${{ github.head_ref }}"
+          elif [[ "${GITHUB_EVENT_NAME}" == "workflow_dispatch" ]]; then 
+            echo "::set-output name=branch::${{ github.event.inputs.branch }}"
+          else
+            echo "::set-output name=branch::INVALID_EVENT_BRANCH_UNKNOWN"
+          fi
+
+  # Parse decompiled data stored on S3 and commit the changes to deadlock-data repo.
+  # This repo is kept separate to keep the deadbot repo diff cleaner, whilst allowing
+  # for easy source control and diff viewing.
+  parse:
+    name: Parse
+    runs-on: ubuntu-latest
+    needs: get-branch
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: deadbot
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install docker-compose
+        run: sudo apt-get install -y docker-compose
+
+      - name: Build Deadbot
+        run: docker-compose build
+        working-directory: deadbot 
+        env:
+          IAM_KEY: ${{ secrets.IAM_KEY }}
+          IAM_SECRET: ${{ secrets.IAM_SECRET }}
+          BOT_WIKI_USER: ${{ secrets.BOT_WIKI_USER }}
+          BOT_WIKI_PASS: ${{ secrets.BOT_WIKI_PASS }}
+      
+      # - name: Run Deadbot
+      #   run: docker-compose up
+      #   working-directory: deadbot 
+
+      - name: Checkout deadlock-data repo
+        uses: actions/checkout@v4
+        with:
+          repository: deadlock-wiki/deadlock-data
+          path: deadlock-data
+          token:  ${{ secrets.GH_TOKEN }}          
+
+      - name: Configure Git
+        working-directory: deadlock-data 
+        run: |
+          git config --global user.email "deadbot1101@gmail.com"
+          git config --global user.name "Deadbot0"
+        
+      - name: Switch to same branch name, create if not found
+        working-directory: deadlock-data
+        run: |
+          git pull origin ${{ needs.get-branch.outputs.branch }} -X ours
+          git switch -c ${{ needs.get-branch.outputs.branch }}
+          git push --set-upstream origin ${{ needs.get-branch.outputs.branch }}
+
+      - name: Pull branch
+        working-directory: deadlock-data 
+        run: git pull origin ${{ needs.get-branch.outputs.branch }} -X ours     
+
+      - name: Copy files to data repo
+        run: cp -r ./deadbot/output-data ./deadlock-data/data/current
```

Comment:
the data repo has some scripts that can assist, you might be able to copy them and mold them to your needs: https://github.com/deadlock-wiki/deadlock-data/blob/develop/Taskfile.yml